packages/ui/src/composites/wui-list-select/index.ts

```diff
@@ -0,0 +1,54 @@
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { globalStyles } from '../../utils/ThemeUtil'
+import '../../components/wui-text'
+import '../wui-wallet-image'
+import '../wui-all-wallets-image'
+import styles from './styles'
+import type { IWalletImage } from '../../utils/TypesUtil'
+
+@customElement('wui-list-select')
+export class WuiListSelect extends LitElement {
+  public static styles = [globalStyles, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @property({ type: Array }) public walletImages?: IWalletImage[] = []
+
+  @property() public imageSrc? = ''
+
+  @property() public name = ''
+
+  @property() public status?: 'installed' | 'qr code' | 'recent'
+
+  @property({ type: Boolean }) public showAllWallets = false
+
+  // -- render ------------------------------------------------------- //
+  public render() {
+    const visual =
```

Comment:
When you have sub-components like this, best create separate methods like:

```ts
private visualTemplate() {
  // ...
  return html`...`
}
```

And then use them inside `render()`. Makes things easier to follow and understand when scanning through code after some time when returning to it.

Reply:
Further down below, you have a relatively complex ternary i.e. `? ... : ... &&`. Those are also hard to follow sometimes, I'd suggest you take out all logic for rendering those into new template functions I suggested above and basically check for all conditions to match before you return `html` or instead just return `null`

This will simplify main render to just

```
// ...
${this.templateAllWallets()}
${this.templateWalletImage()}
// ..
```